tests/MoBi.R.Tests/Services/ProjectTaskSpecs.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using MoBi.Core.Domain.Model;
+using MoBi.R.Services;
+using NUnit.Framework;
+using OSPSuite.BDDHelper.Extensions;
+using static MoBi.R.Tests.DomainHelperForSpecs;
+
+namespace MoBi.R.Tests.Services
+{
+   internal abstract class concern_for_ProjectTask : ContextForIntegration<IProjectTask>
+   {
+      public override void GlobalContext()
+      {
+         base.GlobalContext();
+         sut = Api.GetProjectTask();
+      }
+   }
+
+   internal class when_reading_module_names_from_project : concern_for_ProjectTask
+   {
+      private IReadOnlyList<string> _moduleNames;
+      private MoBiProject _project;
+
+      protected override void Context()
+      {
+         base.Context();
+
+         var projectFile = TestFileFullPath("SampleProject.mbp3");
+         _project = sut.GetProject(projectFile);
+      }
+
+      protected override void Because()
+      {
+         _moduleNames = sut.GetModuleNames(_project);
+      }
+
+      [Test]
+      public void should_return_module_names()
+      {
+         _moduleNames.ShouldNotBeNull();
+         _moduleNames.Count.ShouldBeEqualTo(2);
+         _moduleNames.ShouldContain("Module1");
+         _moduleNames.ShouldContain("Module2");
+      }
+   }
+
+   internal class when_reading_building_blocks_names_from_module : concern_for_ProjectTask
+   {
+      private IReadOnlyList<string> _buildingBlockNames;
+      private MoBiProject _project;
+
+      protected override void Context()
+      {
+         base.Context();
+
+         var projectFile = TestFileFullPath("SampleProject.mbp3");
+         _project = sut.GetProject(projectFile);
+      }
+
+      protected override void Because()
+      {
+         var moduleNames = sut.GetModuleNames(_project);
+         _buildingBlockNames = sut.GetBuildingBlocksNamesFromModuleName(moduleNames.First());
+      }
+
+      [Test]
+      public void should_return_module_building_blocks_names()
+      {
+         _buildingBlockNames.ShouldNotBeNull();
+         _buildingBlockNames.ShouldContain("Organism");
+         _buildingBlockNames.ShouldContain("Molecules");
+         _buildingBlockNames.ShouldContain("Initial Conditions");
+         _buildingBlockNames.ShouldContain("Reactions");
```

Comment:
You can use `ShouldOnlyContain` which will make sure that those ones are all there, and there are no others.

Reply:
It takes as many strings as you need by using `params`

`
public static void ShouldOnlyContain<T>(this IEnumerable<T> items, params T[] itemsToFind)
`

Reply:
Do the same in the simulation names.